### PR TITLE
add pipeline third party auth

### PR DESCRIPTION
**Pull request by luismorenolopera**
Add a new pipeline that checks if the user belongs to the current host.

### CODE REVIEW

File: eox_tenant/test/test_pipeline.py [hunk 1]
```diff
@@ -43,3 +43,63 @@ def test_superadmin_user_fails(self):
         self.user_mock.is_superuser = True
         with self.assertRaises(EoxTenantAuthException):
             safer_associate_by_email(self.backend_mock, {'email': 'fake@example.com'})
+
+
+class AssociationUserSignUpSoruceTest(TestCase):
+    """
+    TODO
```
Comment (felipemontoya) [inline eox_tenant/test/test_pipeline.py:50]: Please write it

File: eox_tenant/pipeline.py [hunk 2]
```diff
@@ -45,3 +45,39 @@ def safer_associate_by_email(backend, details, user=None, *args, **kwargs):
                 )
             return {'user': users[0],
                     'is_new': False}
+
+
+def safer_associate_by_signupsource(backend, request, details, user=None, *args, **kwargs):
+    """TODO"""
```
[the remark this answers is not in the document]
Comment (luismorenolopera) [reply eox_tenant/pipeline.py]: done

File: eox_tenant/pipeline.py [hunk 3]
```diff
@@ -45,3 +45,39 @@ def safer_associate_by_email(backend, details, user=None, *args, **kwargs):
...
+    if user:
+        return None
+
+    email = details.get('email')
```
[the remark this answers is not in the document]
Comment (luismorenolopera) [reply eox_tenant/pipeline.py]: yes, this makes more sense to me

File: eox_tenant/test/test_pipeline.py (hunk 1 above)
Comment (luismorenolopera) [reply eox_tenant/test/test_pipeline.py]: done

**Comment by morenol** [comment]
@luismorenolopera please rebase and add an entry in the unreleased section of the changelog with the description of this feature.

**Pull request closed without merging**